managedplugin/docker.go

```diff
@@ -119,6 +119,8 @@ func pullDockerImage(ctx context.Context, imageName string, authToken string, te
 			return fmt.Errorf("failed to encode Docker auth config: %v", err)
 		}
 		opts.RegistryAuth = encodedAuth
+	} else if dockerHubAuth != "" {
+		opts.RegistryAuth = dockerHubAuth
```

Comment:
Are we sure this is enough, no `additionalHeaders` needed?

Reply:
Yes. I have tested it out with AWS ECR

Reply:
What is the format of `dockerHubAuth`? Because we're doing `registry.EncodeAuthConfig` a bit before for our own registry for username/password auth

Reply:
It is a base64 encoded string... Will document how to generate it in the CLI docs

Reply:
I thought about passing in the username/auth from the spec, but after looking into the `AuthConfig` object, realized that there are more fields that could be passed in and rather than exposing the entire object I wanted to just allow the user to specify the base64 string that has all of the fields that their specific repo requires